src/System.Management.Automation/help/MUIFileSearcher.cs

```diff
@@ -120,11 +121,25 @@ private string[] GetFiles(string path, string pattern)
             // extra logic to select the files that match the given pattern.
             ArrayList result = new ArrayList();
             string[] files = Directory.GetFiles(path);
+
+            string regexPattern = pattern.Replace(".", @"\.");
+            regexPattern = regexPattern.Replace("*", ".*");
+            regexPattern = regexPattern.Replace("?", ".?");
+
             foreach (string filePath in files)
             {
                 if (filePath.IndexOf(pattern, StringComparison.OrdinalIgnoreCase) >= 0)
                 {
                     result.Add(filePath);
+                    break;
+                }
+                // If the input is pattern instead of string, we need to use Regex expression.
+                if (pattern.Contains("*") || pattern.Contains("?"))
+                {             
+                    if (Regex.IsMatch(filePath, regexPattern))
```

Comment:
I agree. We basically need to implementation `Directory.GetFiles(path, pattern);` in a case-insensitive way, and the code at https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs#L5961 is doing the same thing.